src/python/pants/backend/python/target_types_rules.py

```diff
@@ -143,29 +159,174 @@ async def inject_pex_binary_entry_point_dependency(
 # -----------------------------------------------------------------------------------------------
 
 
+def _classify_entry_points(
+    entry_points: FrozenDict[str, FrozenDict[str, str]]
+) -> Generator[Tuple[bool, str, str, str], None, None]:
+    """Looks at each entry point to see if it is a target address or not.
+
+    Yields tuples: is_target, section, name, ref.
+    """
+    for section, values in entry_points.items():
+        for name, ref in values.items():
+            yield ref.startswith(":") or "/" in ref, section, name, ref
+
+
+@rule(desc="Determining the entry points for a `python_distribution` target", level=LogLevel.DEBUG)
+async def resolve_python_distribution_entry_points(
+    request: ResolvePythonDistributionEntryPointsRequest,
+) -> ResolvedPythonDistributionEntryPoints:
+    field_value = request.entry_points_field.value
+    if field_value is None:
+        return ResolvedPythonDistributionEntryPoints()
+
+    address = request.entry_points_field.address
+    classified_entry_points = list(_classify_entry_points(field_value))
+
+    # Pick out all target addresses up front, so we can use MultiGet later.
+    #
+    # This calls for a bit of trickery however (using the "y_by_x" mapping dicts), so we keep track
+    # of which address belongs to which entry point.
```

Comment:
Stale?

Reply:
No, it still applies. `y_by_x` refers to the `address_by_ref` and `binary_entry_point_by_address` respectively. Maybe I should spell them out instead.